pkg/issuer/acme/http/http_test.go

```diff
@@ -44,29 +45,17 @@ func TestCheck(t *testing.T) {
 	tests := []testT{
 		{
 			name: "should pass",
-			reachabilityTest: func(context.Context, string, string) (bool, error) {
-				return true, nil
+			reachabilityTest: func(context.Context, *url.URL, string) error {
+				return nil
 			},
 			expectedOk: true,
 		},
-		{
-			name: "should fail",
-			reachabilityTest: func(context.Context, string, string) (bool, error) {
-				return false, nil
-			},
-		},
-		{
-			name: "should fail with absorbed error",
-			reachabilityTest: func(context.Context, string, string) (bool, error) {
-				return false, &absorbErr{err: fmt.Errorf("failed")}
-			},
-		},
 		{
 			name: "should error",
-			reachabilityTest: func(context.Context, string, string) (bool, error) {
-				return false, fmt.Errorf("failed")
+			reachabilityTest: func(context.Context, *url.URL, string) error {
+				return fmt.Errorf("failed")
 			},
-			expectedErr: true,
+			expectedErr: false,
```

Comment:
I think the test case names here need updating? `should error` with `expectedErr: false` seems a weird combination to me.

Reply:
Also, is there *ever* a case where `expectedErr == true` now?

Reply:
Having another look at the tests. I think they should be valid, but they might be made simpler now that we aren't testing the OK logic